contracts/cw-governance/src/contract.rs

```diff
@@ -0,0 +1,359 @@
+            execute_update_governance_modules(deps, env, to_add, to_remove)
+        }
+        ExecuteMsg::SetItem { key, addr } => execute_set_item(deps, key, addr),
+        ExecuteMsg::RemoveItem { key } => execute_remove_item(deps, key),
+    }?;
+
+    Ok(response.add_attribute("sender", info.sender))
+}
+
+pub fn execute_proposal_hook(msgs: Vec<CosmosMsg<Empty>>) -> Result<Response, ContractError> {
+    Ok(Response::default()
+        .add_attribute("action", "execute_proposal_hook")
+        .add_messages(msgs))
+}
+
+pub fn execute_update_config(deps: DepsMut, config: Config) -> Result<Response, ContractError> {
+    CONFIG.save(deps.storage, &config)?;
+    // We incur some gas costs by having the config's fields in the
+    // response. This has the benefit that it makes it reasonably
+    // simple to ask "when did this field in the config change" by
+    // running something like `junod query txs --events
+    // 'wasm._contract_address=governance&wasm.name=name'`.
+    Ok(Response::default()
+        .add_attribute("action", "execute_update_config")
+        .add_attribute("name", config.name)
+        .add_attribute("description", config.description)
+        .add_attribute(
+            "image_url",
+            config.image_url.unwrap_or_else(|| "None".to_string()),
+        ))
+}
+
+pub fn execute_update_voting_module(
+    env: Env,
+    module: ModuleInstantiateInfo,
+) -> Result<Response, ContractError> {
+    let wasm = module.into_wasm_msg(env.contract.address);
+    let submessage = SubMsg::reply_on_success(wasm, VOTE_MODULE_UPDATE_REPLY_ID);
+
+    Ok(Response::default()
+        .add_attribute("action", "execute_update_voting_module")
+        .add_submessage(submessage))
+}
+
+pub fn execute_update_governance_modules(
+    deps: DepsMut,
+    env: Env,
+    to_add: Vec<ModuleInstantiateInfo>,
+    to_remove: Vec<String>,
+) -> Result<Response, ContractError> {
+    let module_count = GOVERNANCE_MODULE_COUNT.load(deps.storage)?;
+
+    // Some safe maths.
+    let new_total = module_count
+        .checked_add(to_add.len() as u64)
+        .ok_or(ContractError::Overflow {})?
+        .checked_sub(to_remove.len() as u64)
+        .ok_or(ContractError::Overflow {})?;
+    if new_total == 0 {
+        return Err(ContractError::NoGovernanceModule {});
+    }
+    GOVERNANCE_MODULE_COUNT.save(deps.storage, &new_total)?;
+
+    for addr in to_remove {
+        let addr = deps.api.addr_validate(&addr)?;
+        GOVERNANCE_MODULES.remove(deps.storage, addr);
+    }
+
+    let to_add: Vec<SubMsg<Empty>> = to_add
+        .into_iter()
+        .map(|info| info.into_wasm_msg(env.contract.address.clone()))
+        .map(|wasm| SubMsg::reply_on_success(wasm, GOV_MODULE_REPLY_ID))
+        .collect();
+
+    Ok(Response::default()
+        .add_attribute("action", "execute_update_governance_modules")
+        .add_submessages(to_add))
+}
+
+pub fn execute_set_item(
```

Comment:
Hmm leftover code?

Reply:
Yeah, what does this do? What's the use case?